src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Services/RazorSemanticTokensInfoService.cs

```diff
@@ -24,53 +26,83 @@
 
 namespace Microsoft.AspNetCore.Razor.LanguageServer.Semantic;
 
-internal class RazorSemanticTokensInfoService : IRazorSemanticTokensInfoService
+[Export(typeof(IRazorSemanticTokensInfoService)), Shared]
+[method: ImportingConstructor]
+internal class RazorSemanticTokensInfoService(
+    IRazorDocumentMappingService documentMappingService,
+    LanguageServerFeatureOptions languageServerFeatureOptions,
+    IRazorLoggerFactory loggerFactory,
+    ITelemetryReporter? telemetryReporter)
+    : IRazorSemanticTokensInfoService
 {
     private const int TokenSize = 5;
 
-    private readonly IRazorDocumentMappingService _documentMappingService;
-    private readonly LanguageServerFeatureOptions _languageServerFeatureOptions;
-    private readonly RazorLSPOptionsMonitor _razorLSPOptionsMonitor;
-    private readonly IClientConnection _clientConnection;
-    private readonly ILogger _logger;
+    private readonly IRazorDocumentMappingService _documentMappingService = documentMappingService ?? throw new ArgumentNullException(nameof(documentMappingService));
+    private readonly LanguageServerFeatureOptions _languageServerFeatureOptions = languageServerFeatureOptions ?? throw new ArgumentNullException(nameof(languageServerFeatureOptions));
+    private readonly ILogger _logger = loggerFactory.CreateLogger<RazorSemanticTokensInfoService>();
```

Comment:
>loggerFactory.CreateLogger<RazorSemanticTokensInfoService>() [](http://example.com/codeflow?start=39&length=60)

nit: maybe

_logger = (loggerFactory ?? threw new ArgumentNullException(nameof(loggerFactory)).CreateLogger<RazorSemanticTokenInfoServices>();

Reply:
If logger factory is null, this line would throw anyway, so not sure its worth it?

Reply:
I don't care too much, maybe a little nicer to have the name of the argument captured? Feel free to punt.